doc/exts/pylint_messages.py

```diff
@@ -62,28 +59,62 @@ def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
     initialize_extensions(linter)
 
 
-def _get_message_data(data_path: Path) -> tuple[str, str, str, str]:
-    """Get the message data from the specified path."""
+def _get_example_code(data_path: Path) -> str:
+    """Get the example code from the specified path."""
+    if not data_path.exists():
+        raise AssertionError(
+            f"Documentation examples path {data_path} does not exist. "
+            "Please create it and add an example."
+        )
+
     good_py_path = data_path / "good.py"
     bad_py_path = data_path / "bad.py"
     details_rst_path = data_path / "details.rst"
     related_rst_path = data_path / "related.rst"
-    if not data_path.exists():
-        _create_placeholders(data_path, details_rst_path, good_py_path)
+
     good_code = _get_titled_rst(
         title="Correct code", text=_get_python_code_as_rst(good_py_path)
     )
-    bad_code = _get_titled_rst(
-        title="Problematic code", text=_get_python_code_as_rst(bad_py_path)
-    )
+    # Handle multi-file bad code templates
+    if bad_py_path.exists():
+        bad_code = _get_titled_rst(
+            title="Problematic code", text=_get_python_code_as_rst(bad_py_path)
+        )
+    elif (data_path / "bad").exists():
+        bad_files: list[str] = []
+        for file in (data_path / "bad").iterdir():
```

Comment:
We need to sort here, so bad.py is before bad2.py
```suggestion
        for file in sorted((data_path / "bad").iterdir()):
```

Reply:
Turn out it's not that easy to sort Path 😄 took me embarrassingly long to understand